src/coreclr/nativeaot/System.Private.CoreLib/src/System/Enum.NativeAot.cs

```diff
@@ -101,39 +101,39 @@ internal static unsafe bool TryGetUnboxedValueOfEnumOrInteger(object value, out
             switch (elementType)
             {
                 case EETypeElementType.Char:
-                    result = (ulong)(long)Unsafe.As<byte, char>(ref pValue);
+                    result = (ulong)(long)Unsafe.ReadUnaligned<char>(ref pValue);
```

Comment:
The `pValue` is expected to be aligned properly. This changes in this method are unnecessary deoptimization.